package/kedro_viz/models/flowchart.py

```diff
@@ -496,189 +610,278 @@ def is_tracking_node(self):
 
     def is_preview_node(self):
         """Checks if the current node has a preview"""
-        try:
-            is_preview = bool(self.viz_metadata["preview_args"])
-        except (AttributeError, KeyError):
+        if not (self.viz_metadata and self.viz_metadata.get("preview_args", None)):
             return False
-        return is_preview
+
+        return True
 
     def get_preview_args(self):
         """Gets the preview arguments for a dataset"""
-        return self.viz_metadata["preview_args"]
+        return self.viz_metadata.get("preview_args", None)
 
 
-@dataclass
 class TranscodedDataNode(GraphNode):
-    """Represent a graph node of type DATA"""
+    """Represent a graph node of type DATA
 
-    # whether the data node is a free input
-    is_free_input: bool = field(default=False)
+    Args:
+        layer (Optional[str]): The layer that this transcoded data
+        node belongs to. Defaults to `None`.
+        is_free_input (bool): Determines whether the transcoded data
+        node is a free input. Defaults to `False`.
+        stats (Optional[Dict]): Statistics for the data node
```

Comment:
Nit. If the docstring is multi-line it should be indented, it's a bit hard to read which one is the argument, there should be example in the codebase